python/cudf/io/dlpack.py

```diff
@@ -0,0 +1,80 @@
+# Copyright (c) 2019, NVIDIA CORPORATION.
+
+from cudf.dataframe.dataframe import DataFrame
+from cudf.dataframe.series import Series
+from cudf.dataframe.index import Index
+from cudf.dataframe.column import Column
+from cudf.dataframe.buffer import Buffer
+from cudf.dataframe import columnops
+from cudf.utils import ioutils
+from cudf.bindings import dlpack as cpp_dlpack
+from cudf.bindings.GDFError import GDFError
+
+
+def from_dlpack(pycapsule_obj):
+    """Converts from a DLPack tensor to a cuDF object.
+
+    DLPack is an open-source memory tensor structure:
+    `dmlc/dlpack <https://github.com/dmlc/dlpack>`_.
+
+    This function takes a PyCapsule object which contains a pointer to
+    a DLPack tensor as input, and returns a cuDF object. This function deep
+    copies the data in the DLPack tensor into a cuDF object.
+
+    Parameters
+    ----------
+    pycapsule_obj : PyCapsule
+        Input DLPack tensor pointer which is encapsulated in a PyCapsule
+        object.
+
+    Returns
+    -------
+    A cuDF DataFrame or Series depending on if the input DLPack tensor is 1D
+    or 2D.
+    """
+    try:
+        res, valids = cpp_dlpack.from_dlpack(pycapsule_obj)
+    except GDFError:
+        raise ValueError(
+            "Cannot create a cuDF Object from a DLPack tensor of 0 size"
+        )
```

Comment:
I have a feeling this is concealing any GDFError and calling it a DLPack tensor of 0 size.